Documentation/troubleshooting.md

```diff
@@ -74,6 +74,26 @@ A common problem related to `ServiceMonitor` identification by Prometheus is rel
 
 Note: The `ServiceMonitor` references a `Service` (not a `Deployment`, or a `Pod`), by labels *and* by the port name in the `Service`. This *port name* is optional in Kubernetes, but must be specified for the `ServiceMonitor` to work. It is not the same as the port name on the `Pod` or container, although it can be.
 
+### Debugging Why Resources Are Not Being Picked Up
+The Prometheus Operator will reject invalid resources and not reconcile them in the Prometheus configuration, for instance if the `ServiceMonitor object has a references to a secret which doesn't exist in the cluster. When it happens the Operator emits a Kubernetes event detailing the issue.
+
+To check for events related to a `ServiceMonitor`  object, you can use the following command:
+...
+
+{{< alert icon="👉" text="Prometheus Operator can emit Kubernetes events starting with v0.71.0."/>}}
+
+To check for events related to rejected resources, you can use the following command:
+
+```sh
+kubectl get events --field-selector=involvedObject.name="<name of PodMonitor resource>" -n "<namespace where resource is deployed>"
+```
+
+Use the following metrics to identify rejected resources:
+
+```
+prometheus_operator_managed_resources
+```
```

Comment:
Instead of mentioning the metric, I'd refer to the PrometheusOperatorRejectedResources alerting rule:
```suggestion
If you've deployed the Prometheus Operator using kube-prometheus manifests, the `PrometheusOperatorRejectedResources` alert should fire when invalid objects are detected.
```